airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -118,11 +130,54 @@ def __init__(
         self.dynamodb_scan_kwargs = dynamodb_scan_kwargs
         self.s3_bucket_name = s3_bucket_name
         self.s3_key_prefix = s3_key_prefix
+        self.export_time = export_time
+        self.export_format = export_format
 
-    def execute(self, context: Context) -> None:
-        hook = DynamoDBHook(aws_conn_id=self.source_aws_conn_id)
-        table = hook.get_conn().Table(self.dynamodb_table_name)
+        if self.export_time and self.export_time > datetime.now():
+            raise ValueError("The export_time parameter cannot be a future time.")
 
+    @cached_property
+    def hook(self):
+        """Create DynamoDBHook"""
+        return DynamoDBHook(aws_conn_id=self.source_aws_conn_id)
+
+    def execute(self, context: Context) -> None:
+        if self.export_time:
+            self._export_table_to_point_in_time()
+        else:
+            self._export_entire_data()
+
+    def _export_table_to_point_in_time(self):
+        """
+        Export data from start of epoc till `export_time`. Table export will be a snapshot of the table's
+         state at this point in time.
+        """
+        client = self.hook.conn.meta.client
+        response = client.export_table_to_point_in_time(
+            TableArn=self.dynamodb_table_name,
+            ExportTime=self.export_time,
+            S3Bucket=self.s3_bucket_name,
+            S3Prefix=self.s3_key_prefix,
+            ExportFormat=self.export_format,
+        )
+        credentials = self.hook.get_credentials()
+        waiter = self.hook.get_waiter(
+            CUSTOM_WAITER_NAME,
```

Comment:
(Non-blocking) I appreciate the constant here instead of a magic string, but I think in this case I'd likely inline it.  I feel like we usually use waiter names directly and don't have a problem with it here.  But I'll leave it up to you.

Reply:
Ya, looks like an overkill, since we are just using it at a single place.